pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole.go

```diff
@@ -0,0 +1,249 @@
+				},
+			},
+		},
+	}, nil
+}
+
+func newMDCImageStream(cr *mdcv1alpha1.MobileDeveloperConsole) (*imagev1.ImageStream, error) {
+	return &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      cfg.MDCImageStreamName,
+			Labels:    util.Labels(&cr.ObjectMeta, cfg.MDCImageStreamName),
+		},
+		Spec: imagev1.ImageStreamSpec{
+			Tags: []imagev1.TagReference{
+				{
+					Name: cfg.MDCImageStreamTag,
+					From: &corev1.ObjectReference{
+						Kind: "DockerImage",
+						Name: cfg.MDCImageStreamInitialImage,
+					},
+					ImportPolicy: imagev1.TagImportPolicy{
+						Scheduled: false,
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func newMDCDeploymentConfig(cr *mdcv1alpha1.MobileDeveloperConsole) (*openshiftappsv1.DeploymentConfig, error) {
+	labels := map[string]string{
+		"app":     cr.Name,
+		"service": "mdc",
+	}
+
+	cookieSecret, err := util.GeneratePassword()
+	if err != nil {
+		return nil, errors.Wrap(err, "error generating cookie secret")
+	}
+
+	return &openshiftappsv1.DeploymentConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: openshiftappsv1.DeploymentConfigSpec{
+			Replicas: 1,
+			Selector: labels,
+			Triggers: openshiftappsv1.DeploymentTriggerPolicies{
+				openshiftappsv1.DeploymentTriggerPolicy{
+					Type: openshiftappsv1.DeploymentTriggerOnConfigChange,
+				},
+				openshiftappsv1.DeploymentTriggerPolicy{
+					Type: openshiftappsv1.DeploymentTriggerOnImageChange,
+					ImageChangeParams: &openshiftappsv1.DeploymentTriggerImageChangeParams{
+						Automatic:      true,
+						ContainerNames: []string{cfg.MDCContainerName},
+						From: corev1.ObjectReference{
+							Kind: "ImageStreamTag",
+							Name: cfg.MDCImageStreamName + ":" + cfg.MDCImageStreamTag,
+						},
+					},
+				},
+				openshiftappsv1.DeploymentTriggerPolicy{
+					Type: openshiftappsv1.DeploymentTriggerOnImageChange,
+					ImageChangeParams: &openshiftappsv1.DeploymentTriggerImageChangeParams{
+						Automatic:      true,
+						ContainerNames: []string{cfg.OauthProxyContainerName},
+						From: corev1.ObjectReference{
+							Kind: "ImageStreamTag",
+							Name: cfg.OauthProxyImageStreamName + ":" + cfg.OauthProxyImageStreamTag,
+						},
+					},
+				},
+			},
+			Template: &corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: cr.Name,
+					Containers: []corev1.Container{
+						{
+							Name:            cfg.MDCContainerName,
+							Image:           cfg.MDCImageStreamName + ":" + cfg.MDCImageStreamTag,
+							ImagePullPolicy: corev1.PullAlways,
+							Env: []corev1.EnvVar{
+								{
+									Name:  "NAMESPACE",
+									Value: cr.Namespace,
+								},
+								{
+									Name:  "NODE_ENV",
+									Value: "production",
+								},
+								{
+									Name:  "OPENSHIFT_HOST",
+									Value: "WHAT!", // TODO TODO TODO
```

Comment:
TODO TODO TODO :smile: 

@aliok What is this used for?

@odra any previous experience with this kind of thing?

Reply:
I assume this is used somewhere in ups to use the correct app subdomain host, I just use it as a cr param currently but we might be able to retrieve it using the kubernetes rest client

Reply:
> @odra any previous experience with this kind of thing?

LOL

MDC needs to know the OpenShift host to make requests against.

I would like to merge this PR this way, as I wrote in the PR description.

Reply:
Looks like @camilamacedo86 had a similar problem in the past: https://github.com/kubernetes/client-go/issues/600

Anyway, let's address this separately.